cmd/iam-keys-check/main.go

```diff
@@ -0,0 +1,229 @@
+package main
+
+import (
+	"bytes"
+	"encoding/csv"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"sort"
+	"strings"
+	"time"
+)
+
+import (
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/iam"
+	flag "github.com/jessevdk/go-flags"
+	"github.com/trussworks/truss-aws-tools/internal/aws/session"
+	"go.uber.org/zap"
+)
+
+var timestampFormat = "2006-01-02T15:04:05-07:00"
+
+type stringSet map[string]struct{}
+
+func (s stringSet) len() int {
+	return len(s)
+}
+
+func (s stringSet) has(key string) bool {
+	_, ok := s[key]
+	return ok
+}
+
+func (s stringSet) add(key string) {
+	s[key] = struct{}{}
+}
+
+func (s stringSet) toArray(sortValues bool) []string {
+	arr := sort.StringSlice(make([]string, 0, len(s)))
+	for k := range s {
+		arr = append(arr, k)
+	}
+	if sortValues {
+		arr.Sort()
+	}
+	return arr
+}
+
+func newstringSet() stringSet {
+	return stringSet(map[string]struct{}{})
+}
+
+// Options are the command line options
+type Options struct {
+	Profile         string `short:"p" long:"profile" description:"The AWS profile to use." required:"false" env:"AWS_PROFILE"`
+	Region          string `long:"region" description:"The AWS region to use." required:"false" env:"REGION"`
+	Lambda          bool   `short:"l" long:"lambda" description:"Run as an AWS lambda function." required:"false" env:"LAMBDA"`
+	MaxDays         uint   `long:"days" description:"The number of days." default:"90" env:"MAX_DAYS"`
+	SlackWebhookURL string `long:"slack-webhook-url" description:"The Slack Webhook Url." default:"" env:"SLACK_WEBHOOK_URL"`
+	SlackChannel    string `long:"slack-channel" description:"The Slack channel." default:"" env:"SLACK_CHANNEL"`
+}
+
+var options Options
+var logger *zap.Logger
+
+func rowToMap(header []string, row []string) map[string]string {
+	m := map[string]string{}
+	for i, h := range header {
+		m[strings.ToLower(h)] = row[i]
+	}
+	return m
+}
+
+func getCredentialReport(iamClient *iam.IAM, tries int) (*iam.GetCredentialReportOutput, error) {
+	if tries <= 0 {
+		return &iam.GetCredentialReportOutput{}, errors.New("maxmimum number of tries to get credential report reached")
+	}
+	report, err := iamClient.GetCredentialReport(&iam.GetCredentialReportInput{})
```

Comment:
Why don't you just use the `State` attribute from the [GenerateCredentialReportOutput](https://docs.aws.amazon.com/sdk-for-go/api/service/iam/#GenerateCredentialReportOutput) type rather than looking for error codes?

Reply:
State is a string and error codes are defined, so seemed like better option.